### ISSUE TITLE
change load order preload method

### ISSUE THREAD

**Issue by stalgiag** [processing/p5.js#3536]
This feature request is based on the discussion in #3347. 

Essentially some users may want a p5 sketch to load first when used as an element on a page with a high number of external assets. @calebegg proposed that this could be achieved by initializing p5 with the [DOMContentLoaded](https://developer.mozilla.org/en-US/docs/Web/Events/DOMContentLoaded) event rather than the [load](https://developer.mozilla.org/en-US/docs/Web/Events/load) event which is [the current default](https://github.com/processing/p5.js/blob/master/src/core/init.js#L39). 

[Here is a visual explanation of the difference between the two events.](http://web.archive.org/web/20150405114023/http://ie.microsoft.com/testdrive/HTML5/DOMContentLoaded/Default.html).

What do people think about this as a feature?

**Comment by vedhant** [comment]
@stalgiag I feel we could use ```DOMContentLoaded``` if the sketch is independent of the entire webpage. It would make the sketch load much faster.
However if the sketch requires, for example, an image to load then ```onload``` will have to be used.
How do you suggest we incorporate both?

**Comment by limzykenneth** [comment]
I'm thinking it would either be explicit or implecit. 

Explicit meaning that we choose as a default `DOMContentLoaded` but if a user wants some assets to have already loaded, they would need to specify in some way for the library to use `onload` instead.

Implicit would mean that we will need to somehow detect which case would be the best, such as if the user have code that references assets in the DOM then uses `onload` and vice versa.

**Comment by stalgiag** [comment]
Yes @limzykenneth I was suggesting something explicit but if anyone has any thoughts on how it might be done implicitly that would be really neat. I am not sure how that detection could be written.

**Comment by limzykenneth** [comment]
I have no idea either about how to implement the implicit one (not off the top of my head anyway) but I feel that the explicit route may be a bit much for a beginner to handle if the default is `DOMContentLoaded`.

**Comment by Spongman** [comment]
i would recommend retaining the current default and adding an explicit switch (maybe `p5.useDOMContentLoaded = true'` ?) for those that are aware of the caveats.
